src/jose/Jws.ts

```diff
@@ -0,0 +1,132 @@
+import * as ed25519 from '@noble/ed25519';
+import * as secp256k1 from '@noble/secp256k1';
+import base64url from 'base64url';
+import type JwkEd25519Private from './JwkEd25519Private';
+import type JwkEd25519Public from './JwkEd25519Public';
+import type JwsFlattened from './JwsFlattened';
+import type JwkPrivate from './JwkPrivate';
+import type JwkPublic from './JwkPublic';
+import type JwkSecp256k1Private from './JwkSecp256k1Private';
+import type JwkSecp256k1Public from './JwkSecp256k1Public';
+
+/**
+ * Class containing reusable JWS operations.
+ */
+export default class Jws {
+  /**
+   * Signs the given payload as a JWS.
+   * NOTE: this is mainly used by tests to create valid test data.
+   * @throws {Error} if key given is unsupported.
+   */
+  public static async sign (
+    protectedHeader: object,
+    payload: Buffer,
+    privateKeyJwk: JwkPrivate
+  ): Promise<JwsFlattened> {
+
+    const protectedHeaderString = JSON.stringify(protectedHeader);
+    const protectedHeaderBase64UrlString = base64url.encode(protectedHeaderString);
+    const payloadBase64UrlString = base64url.encode(payload);
+    const signingInputBase64urlString = protectedHeaderBase64UrlString + '.' + payloadBase64UrlString;
+    const signingInputBuffer = Buffer.from(signingInputBase64urlString);
+
+    // This is where we will add support for different algorithms over time.
+    let signatureBuffer: Buffer;
+    if (privateKeyJwk.crv === 'Ed25519') {
+      signatureBuffer = await Jws.signEd25519(signingInputBuffer, privateKeyJwk as JwkEd25519Private);
+    } else if (privateKeyJwk.crv === 'secp256k1') {
+      signatureBuffer = await Jws.signSecp256k1(signingInputBuffer, privateKeyJwk as JwkSecp256k1Private);
+    } else {
+      throw new Error(`unsupported key type ${privateKeyJwk.kty} with curve ${privateKeyJwk.crv}`);
+    }
+
+    const signatureBase64UrlString = base64url.encode(signatureBuffer);
+
+    return {
+      payload   : payloadBase64UrlString,
+      protected : protectedHeaderBase64UrlString,
+      signature : signatureBase64UrlString
+    };
+  }
+
+  /**
+   * Implementation of signing using ED25519.
+   */
+  private static async signEd25519 (
+    signingInputBuffer: Buffer,
+    privateKeyJwk: JwkEd25519Private
+  ): Promise<Buffer> {
+    const privateKeyBuffer = base64url.toBuffer(privateKeyJwk.d);
+    const signatureUint8Array = await ed25519.sign(signingInputBuffer, privateKeyBuffer);
+    const signatureBuffer = Buffer.from(signatureUint8Array);
+    return signatureBuffer;
+  }
+
+  /**
+   * Implementation of signing using SECP256K1.
+   */
+  private static async signSecp256k1 (
+    signingInputBuffer: Buffer,
+    privateKeyJwk: JwkSecp256k1Private
+  ): Promise<Buffer> {
+    const privateKeyBuffer = base64url.toBuffer(privateKeyJwk.d);
+    const signatureUint8Array = await secp256k1.sign(signingInputBuffer, privateKeyBuffer);
+    const signatureBuffer = Buffer.from(signatureUint8Array);
+    return signatureBuffer;
```

Comment:
```suggestion
    return Buffer.from(signatureUint8Array);
```

thoughts?

Reply:
My preference for this pattern is so that I can put a breakpoint on the line of return so that debugging is easier. But not too opinionated.